frontend/src/views/user/heatmap.js

```diff
@@ -62,9 +62,68 @@ class WtHeatmap extends HTMLElement {
       })
       .addTo(map);
 
+    let heatLayer = null;
+    let heatMapData = null;
+    let markers = null;
+    const rerenderHeatMap = () => {
```

Comment:
It would be better to add this as a class method. But doing so might need some more refactoring. So for now, this is okay, I guess. 

Besides that, this looks good.